lib/helpers/rule-test-harness.js

```diff
@@ -98,8 +99,9 @@ module.exports = function generateRuleTests({
       if (localConfig === undefined) {
         return;
       }
+      const severity = localConfig === false ? IGNORE_SEVERITY : ERROR_SEVERITY;
 
-      linter.config.rules[name] = localConfig;
+      linter.config.rules[name] = { config: localConfig, severity };
```

Comment:
We should leverage the same `processRules` to allow us to test both old and new configs.